macaw-semmc/src/Data/Macaw/SemMC/Generator.hs

```diff
@@ -102,7 +102,7 @@ data GenState arch ids s =
 
 emptyPreBlock :: RegState (ArchReg arch) (Value arch ids)
               -> Word64
-              -> MM.MemSegmentOff (RegAddrWidth (ArchReg arch))
+              -> MM.MemSegmentOff (ArchAddrWidth arch)
```

Comment:
`MemSegmentOff (ArchAddrWidth arch)` can be further abbreviated to `ArchSegmentOff arch`. (Similarly for other occurrences of `MemSegmentOff (ArchAddrWidth arch)` in this patch.)

Reply:
I replaced some of those too now.